backend/models/course.model.ts

```diff
@@ -0,0 +1,32 @@
+import { Schema, Document, model } from "mongoose";
+
+export interface Course extends Document {
+  id: string;
+  title: string;
+  description: string;
+  image: string;
+  previewImage: string;
+  lessons: [string];
+}
+
+const CourseSchema: Schema = new Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+  },
+  image: {
+    type: String,
+  },
+  previewImage: {
+    type: String,
+  },
+  lessons: {
+    type: [String],
+    required: true,
+  },
```

Comment:
Isn't the supposed to be a list of objects now?

Reply:
Isn't that modules... I think?

Reply:
Lol, this is totally my bad. We can discuss this further, but for now the current implementation is good. Again, sorry for the confusion everyone